src/installer/corehost/cli/test/nativehost/CMakeLists.txt

```diff
@@ -52,4 +52,6 @@ target_link_libraries(${DOTNET_PROJECT_NAME} nethost)
 # Specify non-default Windows libs to be used for Arm/Arm64 builds
 if (CLR_CMAKE_TARGET_WIN32 AND (CLR_CMAKE_TARGET_ARCH_ARM OR CLR_CMAKE_TARGET_ARCH_ARM64))
     target_link_libraries(${DOTNET_PROJECT_NAME} Advapi32.lib Ole32.lib OleAut32.lib)
+elseif(NOT CLR_CMAKE_TARGET_WIN32)
+    target_link_libraries(${DOTNET_PROJECT_NAME} dl)
```

Comment:
This isn't required in some systems (e.g. FreeBSD, macOS).  You can use `${CMAKE_DL_LIBS}` unconditionally.

Reply:
As noted in the top comment, this is already done in https://github.com/dotnet/runtime/blob/5eda36ed557d789d888647745782b261472b9fa3/src/installer/corehost/cli/common.cmake#L44. When I fixed the native build on FreeBSD few weeks ago, it was compiling.